pkg/frontend/transport/handler.go

```diff
@@ -226,6 +226,8 @@ func (f *Handler) reportQueryStats(r *http.Request, queryString url.Values, quer
 		"fetched_index_bytes", numIndexBytes,
 		"sharded_queries", stats.LoadShardedQueries(),
 		"split_queries", stats.LoadSplitQueries(),
+		"insight", "true",
```

Comment:
I'm finding it hard to justify this in the OSS Mimir rather than in our GEM implementation, but I see all that conversation already happened in https://github.com/grafana/mimir/pull/3647 for the alertmanager so I guess that train has left already.

Can you please at least add a test with a comment explaining why we expect this? Otherwise, this label can be dropped during the next refactor as it looks quite useless to most of the users.